### PR TITLE
allow pkgdir accept paths

### PR DESCRIPTION
**Pull request by Roger-luo**
I find it's quite often that when one uses `pkgdir` one actually wants to get the path of some subdir of the package directory, which as a result doing something like

```julia
joinpath(pkgdir(Foo), "src", "subdir")
```

this PR allows `pkgdir` to take a vararg `paths`, so if one wants to do something like above, can now just write

```julia
pkgdir(Foo, "src", "subdir")
```

and it preserves the original behaviour that you can still just put a module to get the root of a module path.

### CODE REVIEW

**Comment by johnnychen94** [comment]
might need to explain it in the docstring with appropriate compat note.

**Comment by Roger-luo** [comment]
not sure which version this PR will be merged into, I currently mark compatibility for 1.7, but might need to update it to 1.8 it doesn't go into it. I'm not sure why there is ambiguity error, the implementation itself seems fine?

File: base/loading.jl [hunk 1]
```diff
@@ -316,16 +316,20 @@ function pathof(m::Module)
...
-    pkgdir(m::Module)
+    pkgdir(m::Module[, paths::String...])
 
- Return the root directory of the package that imported module `m`,
- or `nothing` if `m` was not imported from a package.
- """
-function pkgdir(m::Module)
+Return the root directory of the package that imported module `m`,
+or `nothing` if `m` was not imported from a package. The optional
+argument `paths` can be used to access subdirs of the module root.
```
Comment (IanButterworth) [inline base/loading.jl:323]:
```suggestion
Return the root directory of the package that imported module `m`,
or `nothing` if `m` was not imported from a package. Optionally further
path component strings can be provided to construct a path within the 
package root.

```julia
julia> pkgdir(Foo)
"/path/to/Foo.jl"

julia> pkgdir(Foo, "src", "file.jl")
"/path/to/Foo.jl/src/file.jl"
```

**Review by IanButterworth — CHANGES_REQUESTED** [review]
Great idea. Just a suggestion for the docstring

**Comment by IanButterworth** [comment]
@Roger-luo it seems github suggestions don't play nicely with nested triple quotes. Needs a closing ``` on the example in the docstring

**Comment by giordano** [comment]
>  it seems github suggestions don't play nicely with nested triple quotes. Needs a closing ``` on the example in the docstring

Use quadruple backticks for the suggestion block

**Comment by StefanKarpinski** [comment]
This is very handy and reads nicely too 👍🏻

**Review by IanButterworth — APPROVED** [review]

**Comment by IanButterworth** [comment]
Nice!

**Pull request merged**